src/rdf4cpp/Node.cpp

```diff
@@ -104,16 +104,16 @@ Node::operator std::string() const noexcept {
 }
 
 bool Node::is_literal() const noexcept {
-    return handle_.is_literal();
+    return !handle_.null() && handle_.is_literal();
 }
 bool Node::is_variable() const noexcept {
-    return handle_.is_variable();
+    return !handle_.null() && handle_.is_variable();
 }
 bool Node::is_blank_node() const noexcept {
-    return handle_.is_blank_node();
+    return !handle_.null() && handle_.is_blank_node();
 }
 bool Node::is_iri() const noexcept {
-    return handle_.is_iri();
+    return !handle_.null() && handle_.is_iri();
 }
```

Comment:
Should we also check in the dedicated functions (e.g., Literal::is_literal) if the node is null?

Reply:
I'm not sure, just because
```c++
Literal l{};
l.is_literal() // == false
```
feels kind of stupid

Reply:
I am bit puzzled.
I am actually thinking whether the NULL node should be considered an instance of all types, since it can be joined with anything

Reply:
I like @nkaralis idea of making the null node instance of all types.  ➕ 1️⃣ 
Especially because of it making the `is_xyz` calls consistent across the type hierarchy.

Reply:
we need to consider one thing (discussed with @liss-h): going this route contradicts SPARQL's semantics. In SPARQL, ISIRI(`null`) returns an error.